cmd/build.go

```diff
@@ -77,6 +80,41 @@ func newLocalCommand() *cobra.Command {
 	return cmd
 }
 
+func configVersion(configBytes []byte) (string, bool) {
+	var raw map[string]interface{}
+	if err := yaml.Unmarshal(configBytes, &raw); err != nil {
+		return "", false
+	}
+
+	var configWithVersion struct {
+		Version string "yaml:version"
+	}
+	if err := mapstructure.WeakDecode(raw, &configWithVersion); err != nil {
+		return "", false
+	}
+	return configWithVersion.Version, true
+}
+
+func validateArgs(cmd *cobra.Command, args []string) error {
```

Comment:
Arg validators only receive the non-flag args, not the full list of args, so the filename won't be available from here.

I think this function is generally the right idea, it would just change to accept a `filename string` as arguments and be called from `runExecute`